packages/react-components/src/components/Popover/Popover.tsx

```diff
@@ -113,14 +113,22 @@ export const Popover: React.FC<IPopoverProps> = ({
 
   React.useEffect(() => {
     document.addEventListener('mousedown', handleDocumentClick);
+
+    return () => {
+      document.removeEventListener('mousedown', handleDocumentClick);
+    };
+  }, []);
+
+  React.useEffect(() => {
     if (closeOnEsc) {
       document.addEventListener('keydown', handleHideOnEscape);
     }
     return () => {
-      document.removeEventListener('keydown', handleHideOnEscape);
-      document.removeEventListener('mousedown', handleDocumentClick);
+      if (closeOnEsc) {
```

Comment:
I think when you set `closeOnEsc` dynamically, the handler will not be removed. If I were you, I would always add the handler to `keydown` but return from handler if current value of `closeOnEsc` is false.

Reply:
Great idea! Thanks 👍